backend/src/utils/stream.py

```
@@ -1,173 +1,38 @@
-        
-    if isinstance(message, ToolMessageChunk):
-        data["event"] = "tool_chunk"
-        
-    if isinstance(message, AIMessageChunk):
-        data["event"] = "ai_chunk"
-        data["content"] = message.content[-1].get('text')
-        
-    # logging.debug(f'[utils.stream.handle_types]: {str(data)}')
-    return f"data: {json.dumps(data)}\n\n"
+        logger.exception("Error in astream_chunks", e)
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)